### PR TITLE
Oracle write 4 2024

### PR DESCRIPTION
**Pull request by floptical**
New oracle loading code that relies on this branch: https://github.com/CityOfPhiladelphia/geopetl/tree/oracle-ewkt-fix-4-2024

### CODE REVIEW

File: databridge_etl_tools/oracle/oracle_commands.py [hunk 1]
```diff
@@ -23,3 +24,15 @@ def extract(ctx):
 def extract_json_schema(ctx): 
     """Extracts a dataset's schema in Oracle into a JSON file in S3"""
     ctx.obj.load_json_schema_to_s3()
+
+@oracle.command()
+@click.pass_context
+def truncate_and_load(ctx,):
```
Comment (jrmidkiff) [inline databridge_etl_tools/oracle/oracle_commands.py:30]: Why do we need `truncate_and_load()` and what is it doing that is different from `load()`? Could we delete `truncate_and_load()`?

It does not seem like we have a simple append method though

**Review by jrmidkiff — CHANGES_REQUESTED** [review]
See comment and change request

File: databridge_etl_tools/oracle/oracle.py [hunk 2]
```diff
@@ -264,5 +267,124 @@ def extract(self):
...
+            cursor.execute(copy_stmt)
+            cursor.execute('COMMIT')
+            cursor.execute(f'DROP TABLE {temp_table_name}')
+            cursor.execute('COMMIT')
+            cursor.execute(f'SELECT COUNT(*) FROM {self.table_schema.upper()}.{self.table_name.upper()}')
+            oracle_rows = cursor.fetchone()[0]
+            print(f'assert {num_rows_in_csv} == {oracle_rows}')
+            assert num_rows_in_csv == oracle_rows
+            print('Done.')
+        except Exception as e:
```
Comment (jrmidkiff) [inline databridge_etl_tools/oracle/oracle.py:379]: Eliminate duplicate code - rewrite to: 
```
except (Exception, KeyboardInterrupt) as e: 
    ...
```

File: databridge_etl_tools/oracle/oracle_commands.py (hunk 1 above)
Comment (floptical) [reply databridge_etl_tools/oracle/oracle_commands.py]: reworked as append

**Pull request merged**